Make status bar transparent on android

janicduplessis opened the pull request:
### Details

Make the status bar transparent on Android.

This also fixes 2 issues that were discovered during QA and caused the initial PR (https://github.com/Expensify/App/pull/15778) to be reverted:

1. Splash screen moving. Instead of using StatusBar module from RN to make status bar translucent, use the same code used by RN but directly in MainActivity onCreate. This is needed because JS executes too late and causes the move to happen. See https://github.com/Expensify/App/issues/15770
2. Incorrect window height from RN Dimensions module when using transparent status bar. This is visible on login screen where a bar the size of the statusbar shows at the bottom of the screen. To fix this I used frame from react-native-safe-area-context which has the proper screen size values. Shared in Slack https://expensify.slack.com/archives/C049HHMV9SM/p1678339044151379?thread_ts=1678336100.806819&cid=C049HHMV9SM.

### Fixed Issues

$ https://github.com/Expensify/App/issues/13001  
$ https://github.com/Expensify/App/issues/15770
PROPOSAL: GH_LINK_ISSUE(COMMENT)

### Tests

- [x] Verify that no errors appear in the JS console
- [x] Thorough testing of navigation and status bar on Android
- [x] Ensure status and navigation don't move on app open
- [x] Test various screens and make sure layout looks fine with transparent status bar

### Offline tests

N/A

### QA Steps

- [ ] Verify that no errors appear in the JS console
- [ ] Test various screens and make sure layout looks fine with transparent status bar

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

![image](https://user-images.githubusercontent.com/2677334/224116197-f1c45616-9181-4b52-9c30-d95d258ff15d.png)

<img width="1347" alt="image" src="https://user-images.githubusercontent.com/2677334/223541638-6704d52d-55fb-4d07-b89e-cf16ee00a408.png">

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<img width="365" alt="image" src="https://user-images.githubusercontent.com/2677334/224116960-3e3733e9-956e-4030-8b5f-dfaf8883c4f9.png">

<img width="410" alt="image" src="https://user-images.githubusercontent.com/2677334/223542568-e9d7b895-ee9f-46af-859e-0697c7991604.png">

</details>

<details>
<summary>Mobile Web - Safari</summary>

<img width="387" alt="image" src="https://user-images.githubusercontent.com/2677334/224116425-62af2d19-64be-4486-bde9-dda3f5c0b4cb.png">

<img width="386" alt="image" src="https://user-images.githubusercontent.com/2677334/223542109-a2cb022c-d717-4484-be1e-f80995e288fc.png">

</details>

<details>
<summary>Desktop</summary>

<img width="1202" alt="image" src="https://user-images.githubusercontent.com/2677334/224118287-b85fb58e-98e7-4a7e-9cf1-7fde79340b9f.png">

<img width="1202" alt="image" src="https://user-images.githubusercontent.com/2677334/223542922-c1f3c981-30cf-49a3-884f-9d24ab5288af.png">

</details>

<details>
<summary>iOS</summary>

<img width="390" alt="image" src="https://user-images.githubusercontent.com/2677334/224112702-42388aef-db3c-4549-b640-da09bacdd519.png">

<img width="385" alt="image" src="https://user-images.githubusercontent.com/2677334/223540804-f6f42b63-0cd8-40d1-90cf-56d450ef6b52.png">

<img width="388" alt="image" src="https://user-images.githubusercontent.com/2677334/223540833-82392a29-cea9-408c-9041-f99b12e8a1bb.png">

<img width="390" alt="image" src="https://user-images.githubusercontent.com/2677334/223540975-4e98b68f-8a9d-4c00-9651-d0b448b30e95.png">

<img width="389" alt="image" src="https://user-images.githubusercontent.com/2677334/223541015-b9e4175b-4376-4263-b5c5-a55cc029feb7.png">

<img width="391" alt="image" src="https://user-images.githubusercontent.com/2677334/223541044-b39cde5e-1c1e-44c1-a5b0-92a8a9a4f379.png">

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/2677334/224111135-2f3413db-d436-41cb-9fd6-d9b4ca71d41c.mov

<img width="372" alt="image" src="https://user-images.githubusercontent.com/2677334/224110316-b086d334-60b5-415d-8173-046aeb5af355.png">

<img width="372" alt="image" src="https://user-images.githubusercontent.com/2677334/222912726-2b409ab8-d680-4848-a5f8-d8a2b17fd53c.png">

<img width="371" alt="image" src="https://user-images.githubusercontent.com/2677334/222912805-588d9980-7c20-414d-9f92-74affd6cbcda.png">

<img width="374" alt="image" src="https://user-images.githubusercontent.com/2677334/222912845-eb846f01-d7a7-46ac-8aab-2f5b83889cb4.png">

<img width="372" alt="image" src="https://user-images.githubusercontent.com/2677334/222913304-3cb1405c-5416-4875-9b13-96167d819755.png">

<img width="373" alt="image" src="https://user-images.githubusercontent.com/2677334/222913318-68152e5a-47f5-4422-91a0-c9a11cad2800.png">

</details>

janicduplessis committed:
Make status bar transparent on android

janicduplessis commented on src/pages/signin/SignInPage.js:
```diff
@@ -102,7 +104,7 @@ class SignInPage extends Component {
 
         return (
-            <SafeAreaView style={[styles.signInPage]}>
+            <View style={[styles.signInPage, StyleUtils.getSafeAreaPadding(this.props.insets, 1)]}>
```
For some reason SafeAreaView flickers with the wrong insets on Android, so use insets from withSafeAreaInsets which are fine.

I had to add an extra parameter to `StyleUtils.getSafeAreaPadding` since by default it multiplies bottom and side insets by 0.7 which we don't want here or the size we calculate in `SignInPageLayout` will be wrong

roryabraham replied:
Can we plz include a code comment explaining this and linking to the issue?

janicduplessis commented:
cc @Julesssss @roryabraham

janicduplessis commented:
Added one more fix in SetPassword screen to use SafeAreaView from react-native-safe-area-context instead of react-native.

The flicker I noticed (https://github.com/Expensify/App/pull/15778#discussion_r1131393058) only seems to happen on initial app render, the SafeAreaView on SetPassword screens works fine and doesn't flicker.

Julesssss commented on src/components/CustomStatusBar/index.android.js:
```diff
@@ -9,10 +8,11 @@ import themeColors from '../../styles/themes/default';
-        return <StatusBar />;
+        // Prefer to not render the StatusBar component in Android as it can cause
+        // issues with edge to edge display, which we setup in MainActivity.java.
+        return null;
```
Why even import the library in that case? Lets use native android themes to do whatever `StatusBar.setBarStyle('light-content');` is doing

roryabraham commented on src/components/withWindowDimensions.js:
```diff
@@ -28,67 +28,28 @@ const windowDimensionsProviderPropTypes = {
+    const dimensions = useMemo(() => {
+        const isSmallScreenWidth = frame.width <= variables.mobileResponsiveWidthBreakpoint;
+        const isMediumScreenWidth = frame.width > variables.mobileResponsiveWidthBreakpoint
+      && frame.width <= variables.tabletResponsiveWidthBreakpoint;
```
```suggestion
            && frame.width <= variables.tabletResponsiveWidthBreakpoint;
```

roryabraham commented on src/pages/SetPasswordPage.js:
```diff
@@ -1,10 +1,8 @@
 import React, {Component} from 'react';
-import {
```
Just a thought for a follow-up, we should add a custom lint rule to prevent usage of the `SafeAreaView` from `react-native`

janicduplessis replied:
https://github.com/Expensify/eslint-config-expensify/pull/60

janicduplessis commented:
@roryabraham Updated with your feedback. I added SafeAreaView to /components that just re-export the one from react-native-safe-area-context and removed the old SafeArea component that was unused.

mananjadhav commented:
@janicduplessis @roryabraham @Julesssss  Are we only focus on the status bar or also the bottom bar? For iOS it is transparent at the bottom too. This can be achieved with `windowTranslucentNavigation` but it has its own caveats.


<img width="360" alt="Screenshot 2023-03-13 at 1 31 09 AM" src="https://user-images.githubusercontent.com/3069065/224614109-bb5e5abc-8cc9-4cc3-8508-08c5fa6f9fe4.png">

janicduplessis commented:
@mananjadhav My plan is to start with statusbar and work on navigation bar after. Navigation bar is a bit more tricky since it prevents automatic keyboard handling that we currently rely on on Android. It requires using KeyboardAvoidingView like on iOS. Some code also kind of assume bottom insets if just rounded corners / gesture area like on iOS instead of being a solid bar like on most android devices. So i figured it would be best to address in a follow up.

mananjadhav commented:
Makes sense, and agreed navigation has its fair set of issues to be dealt with especially as it draws the activity behind the status bar.

mananjadhav commented on src/components/withWindowDimensions.js:
```diff
@@ -1,7 +1,7 @@
+import React, {forwardRef, createContext, useMemo} from 'react';
 import PropTypes from 'prop-types';
-import {Dimensions} from 'react-native';
+import {useSafeAreaFrame} from 'react-native-safe-area-context';
```
pardon me for my ignorance, this change will impact all the screens, so are we sure this works well on RNW?

janicduplessis replied:
It has a web implementation, I did test on web and it seems to work well. https://github.com/th3rdwave/react-native-safe-area-context/blob/main/src/NativeSafeAreaProvider.web.tsx.

mananjadhav reviewed (dismissed):
Except for one comment on `withWindowDimensions` support for RNW, I think this looks good.

Other comment on bottom bar is already addressed.

Julesssss reviewed (changes requested):
This is looking better, but unfortunately I'm seeing a lot more jankiness with these changes. Notice the transition between signin/password and show/hide keyboard. We'll need to fix these too.

Julesssss commented:
_Both of these are release builds_

**Current**


https://user-images.githubusercontent.com/10736861/224726911-94134688-b1aa-4a63-b6f5-62489460bb68.mp4


**With changes**


https://user-images.githubusercontent.com/10736861/224726708-935b4f3a-cdf6-4dde-a86d-0a9f998e1dbc.mp4

janicduplessis commented:
Thanks for testing @Julesssss , I suspect this is caused by a different behaviour of frame in react-native-safe-area-context which changes value when keyboard opens vs Dimensions module which doesn’t. I will see how this could be fixed.

janicduplessis commented:
@Julesssss Ok I can confirm that the issue was that the frame returned by react-native-safe-area-context is affected by the keyboard while the one from Dimensions in react-native isn't. The issue was a lot more visible with the magic link since when changing screen in has 2 different keyboards with different heights. This triggers 3 screen change event since it seems like android has to close the text one then open the numeric one.

To avoid this I ended up reverting back to using Dimensions from react-native and just adding top insets on Android to fix the incorrect value.

mananjadhav commented on src/components/withWindowDimensions.js:
```diff
@@ -1,7 +1,7 @@
-/* eslint-disable react/no-unused-state */
-import React, {forwardRef, createContext} from 'react';
+import React, {forwardRef, createContext, useMemo} from 'react';
```
I am going to let @Julesssss @roryabraham confirm if we're okay to use `useMemo` (React hooks)

mananjadhav commented on src/components/withWindowDimensions.js:
```diff
@@ -28,67 +28,33 @@ const windowDimensionsProviderPropTypes = {
-        const {window} = newDimensions;
+    const dimensions = useMemo(() => {
+        // On Android the window height does not include the status bar height, so we need to add it manually.
+        const windowHeight = Platform.OS === 'android'
```
We don't generally allow `Platform.OS` checks. We create [platform specific files](https://github.com/Expensify/App#platform-specific-file-extensions). In this case we could create a util function instead of the whole component that either returns `window.height` or `window.height + insets.top` based on the platform type.

janicduplessis commented:
Update to use platform files

Julesssss reviewed (approved):
Nice work, looking good 👍

Julesssss commented:
This is ready for another review right? CC @roryabraham and @mananjadhav

mananjadhav commented:
Ohh okay thanks for the bump @Julesssss.

> I am going to let @Julesssss @roryabraham confirm if we're okay to use useMemo (React hooks)

I think we're good here with hooks right? do we document any place when we use hooks?

Julesssss commented on src/components/withWindowDimensions/index.js:
```diff
@@ -0,0 +1,76 @@
+    // Is the window width narrow, like on a mobile device?
+    isSmallScreenWidth: PropTypes.bool.isRequired,
+
+    // Is the window width narrow, like on a tablet device?
```
// Is the window width **medium sized**, like on a tablet device?

Julesssss commented:
Ah right, hooks. We do have an existing usage in `BaseHTMLEngineProvider`, but I don't think we've officially approved the usage of hooks yet 😕 

How problematic is it to use an alternative to hooks here?

Julesssss reviewed (dismissed):
Tested on physical devices and it's looking great:
- Pixel 6A (Android 13)
- Nexus 5X (Android 8)

janicduplessis commented:
> How problematic is it to use an alternative to hooks here?

I initially converted this to use hook because react-native-safe-area-context doesn't provide an HOC api for `useSafeAreaFrame`, but it does for `useSafeAreaInsets`. Although converting to hooks reduce complexity of the component a lot since we can also use `useWindowDimensions` from react-native which handles all the subscribing to dimension changes. The hook usage is also not exposed directly, it is just part of the `withWindowDimensions` implementation.

janicduplessis commented:
@Julesssss Updated and tested everything still works.

janicduplessis commented:
Since statusbar is always transparent now I think we can safely remove hacks in Modal

janicduplessis commented:
@aimane-chnaif That would be great, I'm currently looking at the test steps in the fix PR to test it too.

Julesssss commented on src/components/withWindowDimensions/WindowHeightAdjustment.js:
```diff
@@ -0,0 +1,13 @@
+    children: PropTypes.func.isRequired,
+};
+
+const WindowHeightAdjustment = props => props.children(0);
```
What does this do? Is the first child guaranteed to be hight window height data?

janicduplessis replied:
It takes a function for children and passes it an adjustment value, the default implementation passes 0 and the android one passes insets.top. I made this to avoid having isolate the platform specific bits so we don’t have to copy the whole file.

But now that I’m thinking more about it it might be simple to just have a getHeightAdjustment(insets) function that returns 0 by default and insets.top on android.

Julesssss replied:
Thanks! Yeah, that seems a bit more self explanatory to me 👍

mananjadhav replied:
Agreed with the suggestion, but I am still have a question, we have defined `WindowHeightAdjustment` and `WindowHeightAdjustment.android` but only importing `WindowHeightAdjustment.android`?

janicduplessis replied:
Oh I messed up the auto import, updated to use a function and fixed that.

Julesssss commented:
Thanks @aimane-chnaif, are you able to test this? I think it's looking good, and will compile a build today to test the latest changes.

Julesssss commented:
Also @mananjadhav, are you able to take another look at this PR? Thanks!

mananjadhav commented on src/components/withWindowDimensions/getWindowHeightAdjustment.android.js:
```diff
@@ -0,0 +1,5 @@
+// On Android the window height does not include the status bar height, so we need to add it manually.
+export default function getWindowHeightAdjustment(insets) {
```
Should we rename this method? We are not passing any height related info and there's no adjustment made. Could this be `getWindowInsets` or something ?

janicduplessis replied:
I wanted to make the function name more specific to this component use case to avoid confusion where someone would be tempted to use it for something else. Since on all platforms except android it returns 0 and not window insets it might be confusing?

mananjadhav commented on src/components/withWindowDimensions/index.js:
```diff
@@ -1,9 +1,10 @@
-/* eslint-disable react/no-unused-state */
 import React, {forwardRef, createContext} from 'react';
```
What do you think of moving `getWindowHeightAdjustment.js` to `src/libs/getWindowHeightAdjustment/index.js` and `.../index.android.js`, and keeping this as `withWindowDimensions.js` (instead of moving to `withWindowDimensions/index.js`)

janicduplessis replied:
That could work too, although I think this function is very specific to `withWindowDimensions` and should not be used anywhere else, this is why I think I prefer keeping it in the same folder.

mananjadhav reviewed (commented):
Raised some comments.

janicduplessis commented on src/components/withWindowDimensions/getWindowHeightAdjustment.js:
```diff
@@ -0,0 +1,4 @@
+// Some platforms need to adjust the window height.
+export default function getWindowHeightAdjustment() {
```
@Julesssss can you confirm this?

Julesssss replied:
Sorry, I missed this. I agree with @mananjadhav's point about separate files for platform-specific logic.

mananjadhav replied:
I wonder how the current works when we're not importing `getWindowHeightAdjustment.android.js` in the PR.

janicduplessis commented:
@mananjadhav could you have another look? If you don’t agree with what I replied to the review comments let me know and I will make your suggested changes.

roryabraham commented:
I believe @janicduplessis is ooo for the next week or two

janicduplessis commented on src/components/Modal/BaseModal.js:
```diff
@@ -119,18 +119,15 @@ class BaseModal extends PureComponent {
                 animationInTiming={this.props.animationInTiming}
                 animationOutTiming={this.props.animationOutTiming}
-                statusBarTranslucent={this.props.statusBarTranslucent}
+                statusBarTranslucent
```
Is this work merged? Currently I don't find any usages of the `statusBarTranslucent`, this is why I removed the prop as I didn't think it would be useful anymore.

janicduplessis replied:
Sounds good, will add it back for now

OSBotify commented:
Use the links below to test this build in android and iOS. Happy testing!

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/15778/index.html  | https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/15778/index.html  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/15778/index.html)  | ![iOS](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/ios/15778/index.html)  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/15778/NewExpensify.dmg  | https://15778.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/15778/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://15778.pr-testing.expensify.com)  |

Julesssss commented:
Nice work here, it's testing beautifully for me on physical devices new and old 🤩 

- Nexus 5X, Android 8
- Pixel 6a, Android 13

janicduplessis commented:
@aimane-chnaif @Julesssss Conflict fixed!

janicduplessis commented on src/components/Modal/BaseModal.js:
```diff
@@ -127,9 +127,7 @@ class BaseModal extends PureComponent {
-                // When `statusBarTranslucent` is true on Android, the modal fully covers the status bar.
-                // Since `windowHeight` doesn't include status bar height, it should be added in the `deviceHeight` calculation.
-                deviceHeight={this.props.windowHeight + ((this.props.statusBarTranslucent && StatusBar.currentHeight) || 0)}
+                deviceHeight={this.props.windowHeight}
```
I think it should be fine since our device height now always include this extra height from here https://github.com/Expensify/App/pull/15778/files#diff-8c3a1abc2b9785aae309526470e872e7f87b5a7af54cd87d79897b7daf0811b6R3. But yes please let me know if that works, I don’t think I tested this case, only the pdf attachment one

janicduplessis commented:
I verified this doesn't cause regressions for #14563, #15373 and #10352

Julesssss commented:
Ready for full review @aimane-chnaif 👍

Julesssss reviewed (dismissed):
Testing very well fo rme

mananjadhav commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [ ] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/3069065/c5ab457c-2579-44ce-8e11-056cde1e4dee

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/3069065/4623ad47-77a6-4818-9d5d-49c912ed6737

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/3069065/df7093c3-3e4a-4d6d-9b45-ffa5ebc53c71

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/3069065/ddf5b647-86f3-482b-8131-8982d6656db4

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/3069065/f97296bc-c308-4c80-a052-2a68b2f67229

https://github.com/Expensify/App/assets/3069065/996e6189-f2be-4f6c-ae17-7b770fde4136

https://github.com/Expensify/App/assets/3069065/9aec5354-ad35-4ab9-9ae7-0e2125a5679d

<img width="360" alt="android-status-bar" src="https://github.com/Expensify/App/assets/3069065/c0989f76-2972-4ec0-93ac-74213bafa6e2">

</details>


---------

Tests well, I am still yet to test iOS. Xcode is getting stuck at the build stage, I'll upload in a while.

janicduplessis commented on src/libs/getWindowHeightAdjustment/index.android.js:
```diff
@@ -0,0 +1,4 @@
+// On Android the window height does not include the status bar height, so we need to add it manually.
+export default function getWindowHeightAdjustment(insets) {
+    return insets.top;
```
I think it is preferable to use insets from react-native-safe-area-context instead of StatusBar.currentHeight which is a constant so it could have issues if we want to support screen rotation.

If we use insets then we do need platform files as it is implemented on all platform while StatusBar.currentHeight is android only.

Julesssss commented:
Bumping this @janicduplessis, if you could please merge main and push, I'll regenerate the app builds.

Julesssss reviewed (commented):
Kicked off a new build [here](https://github.com/Expensify/App/actions/runs/5059124565)

Julesssss commented:
Hey @aimane-chnaif, what device and OS version is that screencast from?

Julesssss reviewed (commented):
Damn, yeah I'm also seeing this again:

**Android 8**

https://github.com/Expensify/App/assets/10736861/bd5a5c90-7012-4491-b135-70b59a729118

https://github.com/Expensify/App/assets/10736861/5231ca50-1111-4aa3-95be-732b06518c00

janicduplessis commented:
Probably need to make statusbar transparent on splash too, will have a look today.

janicduplessis committed:
Fix new splash screen flicker

janicduplessis committed:
Fix lint

janicduplessis commented:
Oups, there you go!

Julesssss commented:
New CI build [in progress](https://github.com/Expensify/App/actions/runs/5071463555)

OSBotify commented:
:test_tube::test_tube: Use the links below to test this build in android and iOS. Happy testing! :test_tube::test_tube:
| android :robot:  | iOS :apple: |
| ------------- | ------------- |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/android/15778/index.html  | ❌ FAILED ❌  |
| ![Android](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/android/15778/index.html)  | The QR code can't be generated, because the iOS build failed  |
| desktop :computer: | web :spider_web: |
| https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/15778/NewExpensify.dmg  | https://15778.pr-testing.expensify.com  |
| ![Desktop](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://ad-hoc-expensify-cash.s3.amazonaws.com/desktop/15778/NewExpensify.dmg)  | ![Web](https://api.qrserver.com/v1/create-qr-code/?size=120x120&data=https://15778.pr-testing.expensify.com)  |

aimane-chnaif reviewed (approved):
Thanks for the great work!
I think all good to merge

Julesssss commented:
> bar background color and theme changes while on Splash screen

Hey @aimane-chnaif, thanks for retesting. Good spot, but I think we can live with this for now

Julesssss reviewed (approved):
Excellent, thanks again @janicduplessis

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5087958938) to staging by https://github.com/Julesssss in version: 1.3.19-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5115957457) to production by https://github.com/luacmartins in version: 1.3.19-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
